ietf/doc/views_doc.py

```diff
@@ -1179,16 +1155,17 @@ def get_diff_revisions(request, name, doc):
             relateddocument__relationship="replaces",
         )
     )
-    diff_documents.extend(
-        Document.objects.filter(
-            relateddocument__target=doc,
-            relateddocument__relationship="became_rfc"
-        )
-    )
+    if doc.came_from_draft():
+        diff_documents.append(doc.came_from_draft())
```

Comment:
I think you might be hitting the DB once for the `if` and again for the `append()` - better to grab the value and reuse it?

Same applies for `became_rfc()` below (and for both it's possible that Django is clever enough to reuse its query results, but I don't think so)

Reply:
we could memoize came_from_draft and became_rfc later.